src/Tribe/Commerce/Currency.php

```diff
@@ -580,4 +580,256 @@ public function get_symbols_for_codes( $codes ) {
+	 * @param string|null $provider The ticket provider class name.
+	 *
+	 * @return string The thousands separator.
+	 */
+	public function get_currency_thousands_sep( $provider = null ) {
+
+		if ( ! class_exists( $provider ) ) {
+			return $this->get_currency_locale( 'thousands_sep' );
+		}
+
+		if ( 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main' === $provider ) {
+			$decimal_sep = get_option( 'woocommerce_price_thousand_sep' );
+
+			return $decimal_sep;
+		}
+
+		if ( 'Tribe__Tickets_Plus__Commerce__EDD__Main' === $provider && function_exists( 'edd_get_option' ) ) {
+			$decimal_sep = edd_get_option( 'thousands_separator', '.' );
+
+			return $decimal_sep;
+		}
+
+		return $this->get_currency_locale( 'thousands_sep' );
+	}
+
+	/**
+	 * Get the Number of Decimals by provider or default
+	 *
+	 * @since TBD
+	 *
+	 * @param string|null $provider The ticket provider class name
+	 *
+	 * @return string the thousands separator
+	 */
+	public function get_number_of_decimals( $provider = null ) {
+
+		if ( ! class_exists( $provider ) ) {
+			return $this->get_currency_number_of_decimals();
+		}
+
+		if ( 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main' === $provider ) {
+			$decimals = get_option( 'woocommerce_price_num_decimals' );
+
+			return $decimals;
+		}
+
+		if ( 'Tribe__Tickets_Plus__Commerce__EDD__Main' === $provider ) {
+
+			/**
+			 * Filter the Amount of Decimals for EDD
+			 *
+			 * @since TBD
+			 *
+			 * @param int the default number of decimals
+			 */
+			$decimals = apply_filters( 'tribe_edd_format_amount_decimals', 2 );
+
+			return $decimals;
+		}
+
+		return $this->get_currency_number_of_decimals();
+	}
+
+	/**
+	 * Get the Default Amount of Decimals
+	 *
+	 * @since TBD
+	 *
+	 * @return int the amount of decimals
+	 */
+	public function get_currency_number_of_decimals() {
+
+		/**
+		 * Filter the Amount of Decimals
+		 *
+		 * @since TBD
+		 *
+		 * @param int the default number of decimals
+		 */
+		$decimals = apply_filters( 'tribe_format_amount_decimals', 2 );
+
+		return $decimals;
+	}
+
+	/**
+	 * Get the Currency Configuration for all Passed Providers
+	 *
+	 * @since TBD
+	 *
+	 * @param string $provider The ticket provider class name
+	 * @param int    $post_id  The id of the post with tickets
+	 *
+	 * @return array
+	 */
+	public function get_currency_config_for_provider( $providers, $post_id ) {
+
+		if ( ! is_array( $providers ) ) {
+			$providers[] = $providers;
+		}
+
+		$currency = [];
+
+		foreach ( $providers as $provider ) {
+			$currency[ $provider ] = $this->get_currency_by_provider( $post_id, $provider );
+		}
+
+		return $currency;
+	}
+
+	/**
+	 * Get the Currency Formatting Information for a Provider
+	 *
+	 * @since TBD
+	 *
+	 * @param int         $post_id  The id of the post with tickets
+	 * @param string|null $provider The ticket provider class name
+	 *
+	 * @return array an array of formatting details
+	 */
+	public function get_currency_by_provider( $post_id, $provider = null ) {
```

Comment:
We have checks in all these functions for `$provider` - does it make sense to move that here and not call all these functions - just return default values?

Reply:
I think this is probably fine since those functions handle their own defaults pretty well